File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        nums = []
        
        for item in tokens:
            if item.isnumeric():
                nums.append(int(item))
            elif item[0] == "-" and len(item) > 1:
                nums.append(int(item))
            else:
                match item:
                    case "+":
                        nums.append(nums.pop() + nums.pop())
                    case "-":
                        num1 = nums.pop()
                        num2 = nums.pop()
                        nums.append(num2 - num1)
                    case "/":
                        num1 = nums.pop()
                        num2 = nums.pop()
                        nums.append(int(num2 / num1))
                    case "*":
                        nums.append(nums.pop() * nums.pop())

        return nums.pop()

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> n;
        
        for (auto e : tokens) {
            if (isalnum(e[0]) || (e[0] == '-' && e.size() > 1)) {
                n.push(stoi(e));
                
            } else {
                int num2 = n.top();
                n.pop();
                int num1 = n.top();
                n.pop();
                
                switch (e[0]) {
                    case '+':
                        n.push(num1 + num2);
                        break;
                    case '-':
                        n.push(num1 - num2);
                        break;
                    case '*':
                        n.push(num1 * num2);
                        break;
                    case '/':
                        n.push(num1 / num2);
                        break;
                }
            }
        }
        

        
        return n.top();
        
    }
};